src/environment.jl

```diff
@@ -341,3 +341,28 @@ function has_cuda()
         return parse(Bool, flag)
     end
 end
+
+"""
+    MPI.has_rocm()
+
+Check if the MPI implementation is known to have ROCm support. Currently only Open MPI
+provides a mechanism to check, so it will return `false` with other implementations
+(unless overriden).
+
+This can be overriden by setting the `JULIA_MPI_HAS_ROCM` environment variable to `true`
+or `false`.
+"""
+function has_rocm()
+    flag = get(ENV, "JULIA_MPI_HAS_ROCM", nothing)
+    if flag === nothing
+        # Only Open MPI provides a function to check ROCm support
+        @static if MPI_LIBRARY == "OpenMPI"
+            # int MPIX_Query_rocm_support(void)
+            return 0 != ccall((:MPIX_Query_rocm_support, libmpi), Cint, ())
```

Comment:
Why not
```suggestion
            return ccall((:MPIX_Query_rocm_support, libmpi), Bool, ())
```
or even
```suggestion
            return @ccall libmpi.MPIX_Query_rocm_support()::Bool
```
?